src/hooks/useWalletVerification.ts

```diff
@@ -0,0 +1,51 @@
+import { useQuery } from '@tanstack/react-query';
+
+const VERIFY_WALLET_ENDPOINT = `${process.env.NEXT_PUBLIC_BACKEND_URL}/auth/verify-wallet`;
+
+export const useWalletVerification = ({
+  evmAddress,
+  evmMessage,
+  evmSignature,
+  solanaSignature,
+  solanaPublicKey,
+  enabled,
+  queryKey,
+}: {
+  evmAddress?: string;
+  evmMessage?: string;
+  evmSignature?: string;
+  solanaSignature?: string;
+  solanaPublicKey?: string;
+  enabled?: boolean;
+  queryKey?: string;
+}) => {
+  const { data, isSuccess } = useQuery({
```

Comment:
Should be a useMutation instead as we do not want to refetch it (if I understand correctly)